packages/openapi-fetch/test/never-response/never-response.test.ts

```diff
@@ -0,0 +1,143 @@
+import { assertType, describe, expect, test } from "vitest";
+import { createObservedClient } from "../helpers.js";
+import type { components, paths } from "./schemas/never-response.js";
+
+describe("GET", () => {
+  test("sends correct method", async () => {
+    let method = "";
+    const client = createObservedClient<paths>({}, async (req) => {
+      method = req.method;
+      return Response.json({});
+    });
+    await client.GET("/posts");
+    expect(method).toBe("GET");
+  });
+
+  test("sends correct options, returns success", async () => {
+    const mockData = {
+      id: 123,
+      title: "My Post",
+    };
+
+    let actualPathname = "";
+    const client = createObservedClient<paths>({}, async (req) => {
+      actualPathname = new URL(req.url).pathname;
+      return Response.json(mockData);
+    });
+
+    const { data, error, response } = await client.GET("/posts/{id}", {
+      params: { path: { id: 123 } },
+    });
+
+    assertType<typeof mockData | undefined>(data);
+
+    // assert correct URL was called
+    expect(actualPathname).toBe("/posts/123");
+
+    // assert correct data was returned
+    expect(data).toEqual(mockData);
+    expect(response.status).toBe(200);
+
+    // assert error is empty
+    expect(error).toBeUndefined();
+  });
+
+  test("sends correct options, returns undefined on 204", async () => {
+    let actualPathname = "";
+    const client = createObservedClient<paths>({}, async (req) => {
+      actualPathname = new URL(req.url).pathname;
+      return new Response(null, { status: 204 });
+    });
+
+    const { data, error, response } = await client.GET("/posts/{id}", {
+      params: { path: { id: 123 } },
+    });
+
+    assertType<components["schemas"]["Post"] | undefined>(data);
+
+    // assert correct URL was called
+    expect(actualPathname).toBe("/posts/123");
+
+    // assert 204 to be transformed to empty object
+    expect(data).toEqual({});
```

Comment:
Side-note: 

I do not fully agree with this behaviour at https://github.com/openapi-ts/openapi-typescript/blob/main/packages/openapi-fetch/src/index.js#L155. 
1. This makes it impossible to have proper types for 204 codes. 
2. If users of this package DO return some data from 204's (this shouldn't be the case of course, but if their openapi docs DO specify this then it should still return the data from the response instead of transforming it like this)

This PR ensures that a documented 204 response (`content?: never`) will return a `data` as a possible `undefined`. BUT this will now **never** be the case because of this behaviour.

Would love to hear some feedback and if it could be considered to remove this 😄

Reply:
Yes we’re open to revisiting that behavior! Agree that at this point, that does more harm than good. That was an early behavior that just never received much questioning until now.

I’d be open to either of the following:

1. Keep this PR as-is, release as a `patch`, then release that `204` change as a `minor` (breaking) change, OR
2. Include that change in this PR, and release everything as a `minor` (breaking) change

Either way it would constitute a breaking change, but IMO a welcome one. No strong preference on #1 or #2, but #1 would be easier to release faster

Reply:
My preference would also be to keep this PR as-is aswell. I will make a new PR to fix the `204` thing soon 😄 

After that I will have a look at what @zkulbeda suggested for exposing the `status` to help narrowing the `data` type even more 😎

Reply:
That works! Also noticed there’s a tiny conflict with the openapi-typescript-helpers; can merge once that’s resolved.

Also feel free to make an adjustment there if it causes a regression; that recent change was just an arbitrary QoL improvement and didn’t fix any type issues like your PR does

Reply:
Great! Fixed the feedback you gave so everything should be good now 😄